osu.Game/IO/HardLinkHelper.cs

```diff
@@ -29,9 +28,16 @@ public static bool CheckAvailability(string testDestinationPath, string testSour
             try
             {
                 File.WriteAllText(testSourcePath, string.Empty);
-
                 // Test availability by creating an arbitrary hard link between the source and destination paths.
-                return CreateHardLink(testDestinationPath, testSourcePath, IntPtr.Zero);
+
+                bool isHardLinkAvailable = false;
+
+                if (RuntimeInfo.OS == RuntimeInfo.Platform.Windows)
+                    isHardLinkAvailable = CreateHardLink(testDestinationPath, testSourcePath, IntPtr.Zero);
+                else if (RuntimeInfo.OS == RuntimeInfo.Platform.Linux)
+                    isHardLinkAvailable = link(testSourcePath, testDestinationPath) == 0;
```

Comment:
Can this become its own method now that it has multiple platform implementations? Kinda feels bad seeing this same platform choice code in 2 places (here and `RealmFileStore`).